### PR TITLE
Handling Queued Events

### PR DESCRIPTION
**Pull request by dhavalshreyas**
## Queued Events
Queued Events are events sent to the `Sink` synchronously while `Screen`s are being updated. We `queue` them up and are applied after the `Screen` update operation.

## Issues
Our existing implementation of `Queued` events handling has a few issues:

* Each `Workflow` node can handle only one event, if another event is sent, we `fatalError` with a `Action sent to pipe while in the pending state.` message.
* If multiple `Workflow` nodes have events queued, only one of them will be handled, the rest are silently dropped.
* Due to a bug in `enabling` and handling `EventPipes` while handling `queued` events, the Workflow tree can get into an inconsistent state and sometimes leads to a crash.

## Solution
To solve this issue, in this PR, we are not queueing events in the `SubtreeManager`, instead we schedule the event to be delivered to the `EventPipe` asynchronously. This fixes the issues outlined above.

### CODE REVIEW

File: Workflow/Sources/SubtreeManager.swift [hunk 1]
```diff
@@ -315,10 +313,7 @@ extension WorkflowNode.SubtreeManager {
                 fatalError("[\(WorkflowType.self)] Sink sent an action inside `render`. Sinks are not valid until `render` has completed.")
 
             case .pending:
-                validationState = .queued(event)
-
-            case .queued:
-                fatalError("[\(WorkflowType.self)] Action sent to pipe while already in the `queueing` state.")
+                fatalError("[\(WorkflowType.self)] Action sent to pipe while in the `pending` state.")
```
Comment (rjrjr) [inline Workflow/Sources/SubtreeManager.swift:316]: I thought this PR was to eliminate this error?

Comment (dhavalshreyas) [reply Workflow/Sources/SubtreeManager.swift:316]: Yeah, this should not be possible, however the `.pending` state will continue to exist. `fatal` ensures we don't hit this spot.

**Comment by rjrjr** [comment]
Posthoc LGTM

**Review by rjrjr — APPROVED** [review]

**Review by AquaGeek — APPROVED** [review]
Have we run this against the tests in the monorepo?

**Comment by dhavalshreyas** [comment]
> Have we run this against the tests in the monorepo?

Yes, all green.

**Pull request merged**